### PR TITLE
[WIP] Pack and Unpack functions aligned with AMES HIPE

### PR DESCRIPTION
**Pull request by kdenhartog**
This is code to align with [AMES HIPE](https://github.com/hyperledger/indy-hipe/pull/43).

### CODE REVIEW

**Comment by kdenhartog** [comment]
Originally I was going to build on his work, but I diverged from that and made a [commit](https://github.com/hyperledger/indy-sdk/pull/1170/commits/1033002a635700efeed06258bf8db85c40a9a74d) to remove all the microledger code. Would you prefer that I remove all that work and resubmit PR?

**Comment by jovfer** [comment]
yes, we should avoid to include commit history of different subject. Please clean-up the history and update PR

File: libindy/src/services/route/mod.rs [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+use std::rc::Rc;
+
+pub struct RouteService { }
+
+impl RouteService {
+    pub fn new() -> RouteService {
+        RouteService {}
+    }
+
+    pub fn pack_msg(&self, plaintext: &str, recv_keys: &Vec<String>, my_vk: Option<&str>, is_authcrypt: bool,
```
Comment (dbluhm) [inline libindy/src/services/route/mod.rs]: This approach definitely works (awesome job, btw, really looking forward to having this) but I think that separating out auth and anon pack out into separate functions may improve overall understandability of this method. Is there a compelling reason to mix both into the one function?

Comment (kdenhartog) [reply libindy/src/services/route/mod.rs]: when I originally designed it I was combining them. With this new iteration I've been rewriting they are now separate making for much cleaner code to read.

**Comment by kdenhartog** [comment]
more updates added to align with the suggestions brought up. The code needs tested better still.(refactoring the old ones as well as adding new ones). I also need to update the API so the wrappers gain this functionality.

**Comment by jovfer** [comment]
@kdenhartog I suggest to try interactive rebase and drop microledgers commits

**Comment by kdenhartog** [comment]
moved rebased branch to a separate branch called "multiplex-rebase" will close this and open a new PR.

**Comment by jovfer** [comment]
moved to #1210

**Pull request closed without merging**